File: appointments-app-backend/src/application/appointment-service.ts
import { Appointment } from '@/domain/appointment';
import { AppointmentRepository } from '@/infrastructure/appointment-repository';

export class AppointmentService {
    constructor(private appointmentRepository: AppointmentRepository) {}

    async getAppointmentsByPatientId(patientId: string): Promise<Appointment[]> {
        if (!patientId) {
            const error = new Error('Invalid patient ID');
            throw error;
        }

        try {
            const appointments = await this.appointmentRepository.getAppointmentsByPatientId(patientId);
            return appointments;
        } catch (err) {
            throw err; // Re-throw the error to be handled by the caller
        }
    }
}

File: appointments-app-backend-python-poetry/src/application/appointment_service.py
from typing import List
from src.domain.appointment import Appointment
from src.infrastructure.appointment_repository import AppointmentRepository

class AppointmentService:
    def __init__(self, appointment_repository: AppointmentRepository):
        self.appointment_repository = appointment_repository

    def get_appointments_by_patient_id(self, patient_id: str) -> List[Appointment]:
        return self.appointment_repository.get_appointments_by_patient_id(patient_id)
